stdlib/public/runtime/SwiftObject.mm

```diff
@@ -286,6 +286,17 @@ - (BOOL)allowsWeakReference {
 - (BOOL)retainWeakReference {
   return swift_tryRetain(reinterpret_cast<HeapObject*>(self)) != nullptr;
 }
+- (void)_setWeaklyReferenced {
+  auto heapObj = reinterpret_cast<HeapObject *>(self);
+  heapObj->refCounts.setPureSwiftDeallocation(false);
+}
+- (bool)_setAssociatedObject:(id)obj
+                      forKey:(const void *)key
+           associationPolicy:(objc_AssociationPolicy)policy {
+  auto heapObj = reinterpret_cast<HeapObject *>(self);
+  heapObj->refCounts.setPureSwiftDeallocation(false);
+  return false; //let libobjc know that we didn't take care of everything
```

Comment:
This is super nitpicky, but I find this comment to be too vague. I'd suggest, "Let libobjc know that it still needs to take care of setting the associated object" or something like that.

Reply:
Sounds good